### DISCUSSION TITLE
Tab completion support

### DISCUSSION THREAD

**Discussion by autolyticus** [PatrickF1/fzf.fish#190] [Ideas]
Are there any plans for enabling the fuzzy search and completion from getting the output from "complete -C".
Most commands have pretty good completion these days and it would save a lot of time to fill in for example the output of `docker container stop`, or of tmux sessions.

This used to technically be an implemented feature on jethrokuan/fzf but had quite a few bugs with the implementation, especially with fish-shell itself not returning the quoted strings as output of "complete -C": https://github.com/fish-shell/fish-shell/issues/3469

**Comment by mybigman** [comment]
this would definitely be nice and a feature I miss... there's a nice implementation for zsh using https://github.com/romkatv/zsh4humans

its tab completion is really nice especially when cd-ing

**Comment by DevAtDawn** [comment]
something like this? https://github.com/DevAtDawn/fish-fzf-tab
```
function clia
    read -l line; commandline -a $line
    # commandline -a 
end

bind -M insert \et 'fuzzy_complete'

function fuzzy_complete
    complete -C | sort -u | fzf --height 40% --multi --reverse -q (commandline -t) | cut --output-delimiter ' ' -f1 | sed 's/-//g' | clia
    commandline -f end-of-line 
end

```

**Comment by leana8959** [comment]
@DevAtDawn Thanks for the script!  
I'm not sure why, but your script doesn't work on my setup directly (macOS Ventura/fish3.6).  
Using your script as a blueprint, I made mine too!

```fish
# $XDG_CONFIG/fish/config.fish 
...
# Replace fish's default tab pager with fuzzy_complete powered by fzf
bind \t fuzzy_complete
...
```

```fish
# $XDG_CONFIG/fish/functions/fuzzy_complete.fish
function fuzzy_complete
    set -f selected (complete -C | fzf -q (commandline -t) | cut -f1)

    commandline --function repaint
    commandline --current-token --replace -- (string escape -- $selected)
end
```

I'm completely a newbie in fish scripting, feel free if y'all want to tweak it!

Here's an example:
<img width="1202" alt="Capture d’écran 2023-03-30 à 22 06 36" src="https://user-images.githubusercontent.com/87855546/228952975-30fd26be-a6f8-4219-b0c6-f5d48bbd877e.png">

**Comment by PatrickF1** [comment]
Check this out everyone! https://github.com/PatrickF1/fzf.fish/pull/293
Could people upvote this so it's more prominent?

**Reply by leana8959** [reply]
Oh that is the only open pull request :P
Thanks for the heads-up, I didn't notice! (I upvoted :) )

**Comment by PatrickF1** [comment]
After giving a serious try to implement this, I decide to not ever do this. See [my final comment in ](https://github.com/PatrickF1/fzf.fish/pull/293#issuecomment-1793652344).

**Comment by PatrickF1** [comment]
Re-opening so it's more discoverable
